src/Sylius/Bundle/CoreBundle/Resources/config/validation.xml

```diff
@@ -152,4 +152,22 @@
         </property>
     </class>
 
+    <class name="Sylius\Component\Core\Model\Taxon">
+        <property name="images">
+            <constraint name="Valid" />
+        </property>
+    </class>
+
+    <class name="Sylius\Component\Core\Model\TaxonImage">
+        <constraint name="Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity">
+            <option name="fields">
+                <value>code</value>
+                <value>owner</value>
+            </option>
```

Comment:
And we already have this constraint: https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/CoreBundle/Resources/config/doctrine/model/TaxonImage.orm.xml#L21